astropy/convolution/src/convolve.c

```diff
@@ -25,6 +25,9 @@
  */
 
 
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
```

Comment:
This change is definitely right and the last line makes sense given the description at https://docs.scipy.org/doc/numpy-1.17.0/reference/c-api.array.html. But perhaps put it in the `.h` file just before the include of ndarrayobject? (l.42)

Reply:
I've moved these to ``convolve.h``